storage_controller/src/service.rs

```diff
@@ -975,11 +979,20 @@ impl Service {
                 break;
             }
 
-            let client = PageserverClient::new(
+            let client = match PageserverClient::new(
                 node.get_id(),
                 node.base_url(),
                 self.config.pageserver_jwt_token.as_deref(),
-            );
+                self.config.ssl_ca_cert.clone(),
+            ) {
+                Ok(client) => client,
+                Err(e) => {
+                    tracing::error!(
+                        "Failed to create client to detach unknkown shard {tenant_shard_id} on pageserver {node_id}: {e}"
+                    );
+                    return;
```

Comment:
I think this should continue, not return. Even though this is likely to fail for other shards too, we don't know what the cause of the error is, and shouldn't make that assumption. It could prevent cleanup from running entirely.